src/perf/isolates.md

```diff
@@ -0,0 +1,347 @@
+---
+title: Concurrency and isolates
+description: Multithreading in Flutter using Dart isolates.
+---
+
+<?code-excerpt path-base="development/concurrency/isolates/"?>
+
+All Dart code runs in [isolates]({{site.dart-site}}/language/concurrency),
+which are similar to threads,
+but differ in that isolates have their own isolated memory.
+They do not share state in any way,
+and can only communicate by messaging.
+By default,
+Flutter apps do all of their work on a single isolate –
+the main isolate.
+In most cases, this model allows for simpler programming and
+is fast enough that the application's UI doesn't become unresponsive.
+
+Sometimes though, 
+applications need to perform exceptionally large computations
+that can cause "UI jank" (jerky motion).
+If your app is experiencing jank for this reason,
+you can move these computations to a helper isolate.
+This allows the underlying runtime environment 
+to run the computation concurrently
+with the main UI isolate's work
+and takes advantage of multi-core devices.
+
+Each isolate has its own memory
+and its own event loop.
+The event loop processes
+events in the order that they're added to an event queue.
+On the main isolate,
+these events can be anything from handling a user tapping in the UI,
+to executing a function,
+to painting a frame on the screen.
+The following figure shows an example event queue
+with 3 events waiting to be processed.
+
+![The main isolate diagram]({{site.url}}/assets/images/docs/development/concurrency/basics-main-isolate.png){:width="50%"}
+
+For smooth rendering,
+Flutter adds a "paint frame" event to the event queue
+60 times per second(for a 60Hz device).
+If these events aren't processed on time,
+the application experiences UI jank,
+or worse,
+become unresponsive altogether.
+
+![Event jank diagram]({{site.url}}/assets/images/docs/development/concurrency/event-jank.png){:width="50%"}
+
+Whenever a process can't be completed in a frame gap,
+or the time between two frames,
```

Comment:
I would remove the "or" since it isn't clear you are describing the frame gap like this.